app/controllers/ContentController.java

```diff
@@ -0,0 +1,50 @@
+package controllers;
+
+import org.joda.time.DateTime;
+import play.Logger;
+import play.inject.ApplicationLifecycle;
+import services.ThreadNotification;
+import services.content.DataCollectionRunner;
+import services.content.JavaDataCollector;
+import services.dataAccess.AbstractDataAccess;
+import services.sorting.SortingNode;
+import services.sources.TwitterSource;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public class ContentController {
+
+    private Thread sorter;
+    private List<Thread> collectors;
+    private ThreadNotification sortNotification;
+
+    @Inject
+    public ContentController(AbstractDataAccess access, ApplicationLifecycle appLifecycle) {
+        Logger.info("ContentController: Starting application at " + DateTime.now().toString());
+        sortNotification = new ThreadNotification();
+
+
+        sorter = new Thread(new SortingNode(access, sortNotification));
+        sorter.start();
+
+        collectors = new ArrayList<>();
+        Thread twitter = new Thread(new DataCollectionRunner(new JavaDataCollector(access, new
+                TwitterSource()), sortNotification)); // TODO THIS IS AWFUL
```

Comment:
If I'm not mistaken, this is still awful. May be slightly better (at the very least easier to read) to make local instances and pass them in?